bindings/node/src/mongocrypt.cc

```diff
@@ -528,64 +528,89 @@ Value MongoCrypt::MakeExplicitEncryptionContext(const CallbackInfo& info) {
+
+            if (!rangeOptions.IsBuffer()) {
+                throw TypeError::New(Env(), "`rangeOptions` must be a Buffer");
             }
-        }
 
-        if (options.Has("algorithm")) {
-            std::string algorithm = options.Get("algorithm").ToString();
+            std::unique_ptr<mongocrypt_binary_t, MongoCryptBinaryDeleter> binary(BufferToBinary(rangeOptions.As<Uint8Array>()));
+            if (!mongocrypt_ctx_setopt_algorithm_range(context.get(), binary.get())) {
+                throw TypeError::New(Env(), errorStringFromStatus(context.get()));
+            }
+        } else {
             if (!mongocrypt_ctx_setopt_algorithm(
                     context.get(), algorithm.c_str(), algorithm.size())) {
                 throw TypeError::New(Env(), errorStringFromStatus(context.get()));
             }
         }
+    }
 
-        if (options.Has("contentionFactor")) {
-            Napi::Value contention_factor_value = options["contentionFactor"];
-            int64_t contention_factor = contention_factor_value.IsBigInt() ?
-                contention_factor_value.As<BigInt>().Int64Value(nullptr) :
-                contention_factor_value.ToNumber().Int64Value();
-            if (!mongocrypt_ctx_setopt_contention_factor(context.get(), contention_factor)) {
-                throw TypeError::New(Env(), errorStringFromStatus(context.get()));
-            }
+    if (options.Has("contentionFactor")) {
+        Napi::Value contention_factor_value = options["contentionFactor"];
+        int64_t contention_factor = contention_factor_value.IsBigInt() ?
+            contention_factor_value.As<BigInt>().Int64Value(nullptr) :
+            contention_factor_value.ToNumber().Int64Value();
+        if (!mongocrypt_ctx_setopt_contention_factor(context.get(), contention_factor)) {
+            throw TypeError::New(Env(), errorStringFromStatus(context.get()));
         }
+    }
 
-        if (options.Has("queryType")) {
-            std::string query_type_str = options.Get("queryType").ToString();
-            if (!mongocrypt_ctx_setopt_query_type(context.get(), query_type_str.data(), -1)) {
-                throw TypeError::New(Env(), errorStringFromStatus(context.get()));
-            }
+    if (options.Has("queryType")) {
+        std::string query_type_str = options.Get("queryType").ToString();
+        if (!mongocrypt_ctx_setopt_query_type(context.get(), query_type_str.data(), -1)) {
+            throw TypeError::New(Env(), errorStringFromStatus(context.get()));
         }
     }
 
     std::unique_ptr<mongocrypt_binary_t, MongoCryptBinaryDeleter> binaryValue(BufferToBinary(valueBuffer.As<Uint8Array>()));
-    if (!mongocrypt_ctx_explicit_encrypt_init(context.get(), binaryValue.get())) {
+
+    const bool isExpressionMode = options.Has("expressionMode") && options.Get("expressionMode").IsBoolean()
+                                ? options.Get("expressionMode")
+                                : false;
```

Comment:
This should also work and might be a bit more natural:

```suggestion
    const bool isExpressionMode = options.Get("expressionMode").ToBoolean();
```

(corresponding to `!!options.expressionMode` in JS terms)

Reply:
the documentation I read says that Object::get() returns undefined if the key is not present. But I'm now realizing that probably means a Value containing an undefined 🤦 

I'll try this out, thanks